src/contracts/core/Slasher.sol

```diff
@@ -27,7 +28,218 @@ import "@openzeppelin-upgrades/contracts/proxy/utils/Initializable.sol";
+        // TODO: events!
+    }
+
+    // Mapping: Operator => Strategy => epoch => AVS => requested slashed bips
+    mapping(address => mapping(IStrategy => mapping(int256 => mapping(address => uint256)))) public requestedSlashedBips;
+
+    // Mapping: Operator => Strategy => epoch => pending slashed amount, where pending slashed amount is the
+    // amount that will be slashed when slashing is executed for the current epoch, assuming no existing requests are cancelled or nullified. summed over all AVSs
+    mapping(address => mapping(IStrategy => mapping(int256 => uint256))) public pendingSlashedBips;
+
+    // @notice fetches the bips slashable by an AVS for the shares of a certain strategy delegated to a certain operator for a certain epoch
+    function getSlashableBips(
+        address operator, 
+        address avs, 
+        IStrategy strategy, 
+        int256 epoch
+    ) public pure returns (uint256) {
+        // TODO: this is a stub. it needs implementation
+        return 10;
+    }
+
+    // @notice fetches the maximum slashing rate -- expressed in bips -- by an AVS for the shares of a certain strategy for a certain epoch
+    function getMaxSlashingRate(
+        address avs,
+        IStrategy strategy, 
+        int256 epoch
+    ) public pure returns (uint256) {
+        // TODO: this is a stub. it needs implementation
+        return 10000;
+    }
+
+    function createSlashingRequest(
+        address operator,
+        IStrategy[] memory strategies,
+        uint256 bipsToSlash
+    ) external {
+        require(bipsToSlash <= SlashingAccountingUtils.BIPS_FACTOR, "invalid slashing amount");
+        require(bipsToSlash != 0, "cannot slash 0");
+        address avs = msg.sender;
+        int256 epoch = SlashingAccountingUtils.currentEpoch();
+        for (uint256 i = 0; i < strategies.length; ++i) {
+            IStrategy strategy = strategies[i];
+            uint256 requestedSlashedBipsBefore = requestedSlashedBips[operator][strategy][epoch][avs];
+            uint256 requestedSlashedBipsAfter = requestedSlashedBipsBefore + bipsToSlash;
+            // TODO: consider loss of precision here. this is undesirable; we should fix it
+            uint256 bipsAllowedToSlash =
+                getSlashableBips(operator, avs, strategy, epoch) * getMaxSlashingRate(avs, strategy, epoch) / SlashingAccountingUtils.BIPS_FACTOR;
+            int256 changeInPendingSlashedBips = _changeInPendingSlashedBips({
+                bipsAllowedToSlash: bipsAllowedToSlash,
+                requestedSlashedBipsBefore: requestedSlashedBipsBefore,
+                requestedSlashedBipsAfter: requestedSlashedBipsAfter
+            });
+
+            // TODO: events
+            requestedSlashedBips[operator][strategy][epoch][avs] = requestedSlashedBipsAfter;
+            pendingSlashedBips[operator][strategy][epoch] = uint256(
+                int256(pendingSlashedBips[operator][strategy][epoch]) + int256(changeInPendingSlashedBips)
```

Comment:
if we're converting to int256 here, might as well make bipsToSlash an int256 everywhere to make sure we don't hit overflow

Reply:
currently we are avoiding overflow by constraining inputs to `createSlashingRequest` be <= 10000 bips, so it's impossible to send enough transactions to overflow.
I think this is a reasonable suggestion though!